Sources/TuistDependencies/Mappers/ExternalProjectsPlatformNarrowerGraphMapper.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import TuistCore
+import TuistGraph
+
+/**
+ When Swift Packages don't declare the platforms that they support, the Swift Package Manager defaults the value
+ to 'support all the platforms'. This default behaviour is inherited into the Xcode projects that we generate off the packages
+ and that causes compilation issues. Xcode must resolve this issue at build-time by cascading the platform requirements
+ down from nodes in the graph that are closer to the root. This is a behaviour that we need to copy over to Tuist. In our case
+ the logic is executed at generation time.
+ */
+public final class ExternalProjectsPlatformNarrowerGraphMapper: GraphMapping {
```

Comment:
not directly related to this PR: For utilities and mappers like this, should we default to using a `struct` if we don't hold any state?

Reply:
Yes, they don't need to be class. I'll update them